core/views.py

```diff
@@ -45,69 +59,90 @@ def home(request):
     return render(request, "home.html", context)
 
 
-@ratelimit(key="user_or_ip", rate="1/m", block=True)
+# @ratelimit(key="user_or_ip", rate="1/m", block=True)
 def addHackathons(request):
+    print(request.method)
     if request.method == "POST":
         form = HackathonForm(request.POST)
+        print("got the post requets")
         if form.is_valid():
+            print("sending the form")
+            form.save(commit=False)
+            form.review_status = ReviewStatus.Pending
+            form.created_by = Hacker.objects.get(id=request.user.id)
+            form.save()
             return redirect("home")
+        else:
+            print(form.errors)
     else:
         form = HackathonForm()
     return render(request, "hackathons/add_hackathon.html", {"form": form})
 
 
-# def calendar(request):
-#     context = {
-#         "title": "Add a New Hackathon",
-#         "content": "Use this form to add a new hackathon to the database.",
-#     }
-#     return render(request, "../templates/calendar.html", context)
-
+def hackathon_page(request):
+    tdy_date = timezone.now()
+    view_type = request.GET.get("view_type")
+    country = request.GET.get("country")
+    city = request.GET.get("city")
+    start = request.GET.get("start")
+    end = request.GET.get("end")
 
-class HackathonsPage(ListView):
-    template_name = "hackathons/hackathons.html"
-    context_object_name = "hackathons"
-    paginate_by = 30
+    user_votes = Vote.objects.filter(from_hacker=request.user.id)
+    user_votes_dict = {
+        vote.hackathon_id.duplication_id: vote.type_vote for vote in user_votes
+    }
+    print(user_votes_dict)
+    query_base = Q(end_date__gte=tdy_date, is_public=True)
+    if country and country != "none" and country != "World":
```

Comment:
much better ways to do this but if it works it passess

Reply:
we will def have a better system for v2